lib/Sema/ConstraintSystem.cpp

```diff
@@ -3516,10 +3516,26 @@ void ConstraintSystem::resolveOverload(ConstraintLocator *locator,
   if (isDebugMode()) {
     PrintOptions PO;
     PO.PrintTypesForDebugging = true;
-    llvm::errs().indent(solverState ? solverState->depth * 2 : 2)
-      << "(overload set choice binding "
-      << boundType->getString(PO) << " := "
-      << adjustedRefType->getString(PO) << ")\n";
+
+    if (!OpenedTypes.empty()) {
+      auto &log = llvm::errs();
+      log.indent(solverState ? solverState->depth * 2 : 2);
+      log << "(overload set choice binding " << boundType->getString(PO)
+          << " := " << adjustedRefType->getString(PO);
+      for (const auto &opened : OpenedTypes) {
```

Comment:
This is not actually correct - `OpenedTypes` store information across whole constraint system, we want to print only generic parameters that are relevant to this overload choice (identified by a locator).